PalasoUIWindowsForms/ImageToolbox/ArtOfReadingChooser.cs

```diff
@@ -218,10 +227,98 @@ private void ArtOfReadingChooser_Load(object sender, EventArgs e)
+			if (_searchLanguageMenu.Visible)
+				searchLang = _searchLanguageMenu.ToolTipText;
+			msg += Environment.NewLine + Environment.NewLine +
+					String.Format("The search box is currently set to {0}".Localize("ImageToolbox.SearchLanguage"), searchLang);
+			if (PlatformUtilities.Platform.IsWindows && !_searchLanguageMenu.Visible)
+			{
+				msg += Environment.NewLine + Environment.NewLine +
+						"Did you know that there is a new version of this collection which lets you search in Arabic, Bengali, Chinese, English, French, Indonesian, Hindi, Portuguese, Spanish, Thai, or Swahili?  It is free and available for downloading."
+							.Localize("ImageToolbox.NewMultilingual");
+				betterLinkLabel1.Visible = true;
+				betterLinkLabel1.BackColor = Color.White;
+				betterLinkLabel1.Location = new Point(betterLinkLabel1.Left, _messageLabel.Bottom + 4);
+			}
+			_messageLabel.Text = msg;
+		}
+
+		protected class LanguageChoice
+		{
+			static readonly List<string> _languageNamesRecognizableByOtherLatinScriptReaders = new List<string> { "en", "fr", "es", "it", "tpi", "pt", "id" };
+			private readonly CultureInfo _info;
+
+			public LanguageChoice(CultureInfo ci)
+			{
+				_info = ci;
+			}
+
+			public string Id { get { return _info.Name == "zh-Hans" ? "zh" : _info.Name; } }
+
+			public string NativeName { get { return _info.NativeName; } }
+
+			public override string ToString()
+			{
+				if (_info.NativeName == _info.EnglishName)
+					return _info.NativeName;	// English (English) looks rather silly...
+				if (_languageNamesRecognizableByOtherLatinScriptReaders.Contains(Id))
+					return _info.NativeName;
+				return String.Format("{0} ({1})", _info.NativeName, _info.EnglishName);
+			}
+		}
+
+		private void SetupSearchLanguageChoice()
+		{
+			var indexLangs = _imageCollection.IndexLanguageIds;
+			if (indexLangs != null)
+			{
+				var count = 0;
+				foreach (var id in indexLangs)
+				{
+					if (count == 0)
+						_searchLanguageMenu.Visible = true;	// do this once.
+					var ci = id == "zh" ? new CultureInfo("zh-Hans") : new CultureInfo(id);
+					var choice = new LanguageChoice(ci);
+					var item = _searchLanguageMenu.DropDownItems.Add(choice.ToString());
+					item.Tag = choice;
+					item.Click += SearchLanguageClick;
+					if (id == SearchLanguage)
+					{
+						_searchLanguageMenu.Text = choice.NativeName;
+						_searchLanguageMenu.ToolTipText = choice.ToString();
+					}
+					++count;
+				}
+			}
+		}
+
+		void SearchLanguageClick(object sender, EventArgs e)
+		{
+			var item = sender as ToolStripItem;
+			if (item != null)
+			{
+				var lang = item.Tag as LanguageChoice;
+				if (lang != null && SearchLanguage != lang.Id)
+				{
+					_searchLanguageMenu.Text = lang.NativeName;
+					_searchLanguageMenu.ToolTipText = lang.ToString();
+					SearchLanguage = lang.Id;
+					_imageCollection.ReloadImageIndex(lang.Id);
+					SetMessageLabelText();		// Update with new language name.
```

Comment:
The tooltip is being updated if you look.  We don't have the LanguageChoice object in both places where SetMessageLabelText() is called.  And if we're using the old index, we never have one of those.  I think the current code is working okay.